Gems/LyShine/Code/Source/UiTextComponent.cpp

```diff
@@ -3381,6 +3381,15 @@ void UiTextComponent::Activate()
     FontNotificationBus::Handler::BusConnect();
     LanguageChangeNotificationBus::Handler::BusConnect();
 
+#if defined(CARBONATED)
+    // Reload the font families if they were deleted while this UiTextComponent was inactive.
+    // It is important for UiTextComponent which can be activated and deactivated many times.
+    if (!m_fontFamily->normal || !m_fontFamily->bold || !m_fontFamily->italic || !m_fontFamily->boldItalic)
```

Comment:
The condition must be &&, because it frees & nullptrs (SAFE_RELEASE) them all.

Reply:
@aefimov3 please clarify, I misunderstand your idea.
If you means (!(A && B && C)), it is the same with (!A || !B || !C).

Reply:
If font family is released then all the members are nullptrs. So the condition should be if **all** the members are nullptr then reload. Your current condition is: if **any** member is nullptr then reload. I'm not sure if there can be a valid font with some members nullptr, I guess it might. && condition works for such case.

Reply:
if (!m_fontFamily->normal && !m_fontFamily->bold && !m_fontFamily->italic && !m_fontFamily->boldItalic)

Reply:
Understand.
My rule is more stronger.
For example, if some fonts in family are released and some are not released, my rule still will work.
From the other side, it is enough to check only one any font in family. Result will be the same.